redisinsight/ui/src/pages/rdi/pipeline/components/refresh-pipeline-popover/RefreshPipelinePopover.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react'
+import { useParams } from 'react-router-dom'
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiIcon,
+  EuiPopover,
+  EuiText,
+  EuiSpacer,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui'
+
+import { useDispatch, useSelector } from 'react-redux'
+import { sendEventTelemetry, TelemetryEvent } from 'uiSrc/telemetry'
+import { fetchRdiPipeline, rdiPipelineSelector } from 'uiSrc/slices/rdi/pipeline'
+
+import styles from './styles.module.scss'
+
+const RefreshPipelinePopover = () => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false)
+
+  const { loading, data } = useSelector(rdiPipelineSelector)
+
+  const { rdiInstanceId } = useParams<{ rdiInstanceId: string }>()
+  const dispatch = useDispatch()
+
+  const handleRefreshClick = () => {
+    setIsPopoverOpen(false)
+    dispatch(fetchRdiPipeline(rdiInstanceId))
+  }
+
+  const handleRefreshWarning = () => {
+    setIsPopoverOpen(true)
+    sendEventTelemetry({
+      event: TelemetryEvent.RDI_PIPELINE_REFRESH_CLICKED,
+      eventData: {
+        id: rdiInstanceId,
+        jobsNumber: data?.jobs?.length || 'none',
```

Comment:
any specific reason we couldn't just pass `0` instead of `'none'`?

Reply:
Not specific, but it is more clear. Because in my opinion there are two cases:
 - we have pipeline, but user removed all jobs. Jobs length = 0
 - we have failed pipeline request, it means there is no pipeline, so none mo clear for me